File: src/main/java/com/vjiki/music/service/SongService.java
package com.vjiki.music.service;

import java.util.List;
import java.util.UUID;

import com.vjiki.music.dto.SongResponse;
import com.vjiki.music.pagination.CursorPageResponse;

public interface SongService {

    List<SongResponse> getSongs(UUID userId);

    CursorPageResponse<SongResponse> getSongsPage(UUID userId, int limit, String cursor);
}

File: bin/main/com/vjiki/music/service/SongService.kt
package com.vjiki.music.service

import com.vjiki.music.dto.SongResponse
import com.vjiki.music.pagination.CursorPageResponse
import java.util.UUID

interface SongService {
    fun getSongs(userId: UUID): List<SongResponse>

    fun getSongsPage(userId: UUID, limit: Int, cursor: String?): CursorPageResponse<SongResponse>
}

